### PR TITLE
Add target for Big-endian ARM Cortex-R4F/R5F MCUs

### PR DESCRIPTION
**Pull request by paoloteti**
The ARM Real-Time (‘R’) profile provides high-performing processors for safety-critical environments.

Cortex-R has ARM, Thumb instruction whereas Cortex-M makes use of Thumb only.

CI/Dockerfile is intentionally in the `disabled` folder.

### CODE REVIEW

File: src/ci/docker/disabled/dist-armebv7r-none-eabihf/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ENV GCC_LINARO=gcc-linaro-7.2.1-2017.11-x86_64_armeb-eabi
+
+RUN curl -sL $BASE_URL/$GCC_LINARO.tar.xz | tar -xJ
+
+ENV PATH=$PATH:/$GCC_LINARO/bin
+
+ENV TARGET=armebv7r-none-eabihf
+
+ENV CC_armebv7r_none_eabihf=armeb-eabi-gcc \
+    CFLAGS_armebv7r_none_eabihf="-mthumb -march=armv7-r"
```
Comment (parched) [inline src/ci/docker/disabled/dist-armebv7r-none-eabihf/Dockerfile]: To match this rust target, you want `-marm` instead of `-mthumb`.

Comment (paoloteti) [reply src/ci/docker/disabled/dist-armebv7r-none-eabihf/Dockerfile]: Cortex-R has ARM+Thumb instruction (dual mode). Wake-up in ARM mode and can use Thumb instr. So here `armv7-r` select ARM mode, `-mthumb` extend support for Thumb.

The same logic also in LLVM: the triple `armebv7r-none-eabihf` and the feature `+thumb2`.
[This](https://github.com/paoloteti/ti-hercules-bsp/blob/master/boards/Makefile.common) is a working Makefile that use `-mthumb -march=armv7-r` and [this](https://github.com/paoloteti/ti-hercules-bsp/blob/master/boards/ti_tms570/armebv7r-none-eabihf.json) a working JSON.

**Comment by alexcrichton** [comment]
To confirm, we've matched naming so far sort of with Debian but otherwise just tried to find preexisting conventions to name targets. In that sense, there's precedent for this name?

**Comment by paoloteti** [comment]
`armeb-linux-gnueabi` is in the Debian [list](https://wiki.debian.org/Multiarch/Tuples) and
`armebv7-eabihf--musl` is a well know [toolchain](https://toolchains.bootlin.com/releases_armebv7-eabihf.html). Similar name is used by Linaro.

File: src/ci/docker/disabled/dist-armebv7r-none-eabihf/Dockerfile (hunk 1 above)
Comment (paoloteti) [reply src/ci/docker/disabled/dist-armebv7r-none-eabihf/Dockerfile]: Ahh ok. Default shall be Arm. I'll move to `-marm` removing `+thumb2` too. 
I'll test again on target (TI TMS570LS3137) in the afternoon. Thanks.

**Comment by paoloteti** [comment]
Updated with comments from @parched

**Pull request merged**